File: app/src/main/java/com/savvasdalkitsis/gameframe/grid/view/GridTouchedListener.java
package com.savvasdalkitsis.gameframe.grid.view;

public interface GridTouchedListener {

    GridTouchedListener NO_OP = new GridTouchedListener() {
        @Override
        public void onGridTouchStarted() {}

        @Override
        public void onGridTouch(int startColumn, int startRow, int column, int row) {}

        @Override
        public void onGridTouchFinished() {}
    };

    void onGridTouchStarted();

    void onGridTouch(int startColumn, int startRow, int column, int row);

    void onGridTouchFinished();
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/feature/workspace/element/grid/view/GridTouchedListener.kt
package com.savvasdalkitsis.gameframe.feature.workspace.element.grid.view

interface GridTouchedListener {

    fun onGridTouchStarted()

    fun onGridTouch(startColumn: Int, startRow: Int, column: Int, row: Int)

    fun onGridTouchFinished()
}
